scheduler/src/cook/mesos/rebalancer.clj

```diff
@@ -218,10 +218,22 @@
   [{:keys [task->scored-task host->spare-resources] :as state}
    {:keys [min-dru-diff safe-dru-threshold] :as params}
    pending-job-ent]
+  ;;We need to rely on this being a priority map TODO check that this is true
+  (when-not (instance? clojure.data.priority_map.PersistentPriorityMap task->scored-task)
+    (log/fatal "Implementation detail failed; needed priority, got" (class task->scored-task))
+    (throw (ex-info "Implementation detail failed" {})))
   (let [{pending-job-mem :mem pending-job-cpus :cpus} (util/job-ent->resources pending-job-ent)
         pending-job-dru (compute-pending-job-dru state pending-job-ent)
+
         ;; This will preserve the ordering of task->scored-task
         host->scored-tasks (->> task->scored-task
+                                ;;TODO maybe we can just change every backfilled task here to have the worst possible DRU
+                                ;;this should bias us towards always killing those tasks first...
+                                (reduce-kv (fn [m k {:keys [task] :as scored-task}]
```

Comment:
We can't make the change there, since that code is used in many places, and we only want to penalize backfilled jobs during preemption, not during normal scheduling.